test/common.py

```diff
@@ -2542,9 +2632,15 @@ def browser_open(cls, url):
       if worker_id is not None:
         # Running in parallel mode, give each browser its own profile dir.
         browser_data_dir += '-' + str(worker_id)
-      if os.path.exists(browser_data_dir):
-        utils.delete_dir(browser_data_dir)
+
+      # Delete old browser data directory. If we cannot (the data dir is in use on Windows),
+      # switch to another dir.
+      while not force_delete_dir(browser_data_dir):
+        browser_data_dir += '-another'
```

Comment:
This seems like it could generated `-another-another-another-another`?    Should we use a counter and do `_1`, `_2` etc?

Also, like the the rest of this PR I'd would love to not run this loop except on windows.

Reply:
Indeed it can, and does. I wanted to keep the logic simple. Updated to use `_number`